Rules/UseUsingScopeModifierInNewRunspaces.cs

```diff
@@ -0,0 +1,430 @@
+
+        private readonly IEnumerable<string> _inlineScriptCmdletNamesAndAliases = Helper.Instance.CmdletNameAndAliases("InlineScript");
+
+        private readonly IEnumerable<string> _foreachObjectCmdletNamesAndAliases = Helper.Instance.CmdletNameAndAliases("Foreach-Object");
+
+        private readonly IEnumerable<string> _invokeCommandCmdletNamesAndAliases = Helper.Instance.CmdletNameAndAliases("Invoke-Command");
+
+        private readonly Dictionary<string, List<VariableExpressionAst>> _varsDeclaredPerSession = new Dictionary<string, List<VariableExpressionAst>>();
+        
+        private readonly List<DiagnosticRecord> _diagnosticAccumulator;
+        
+        private readonly UseUsingScopeModifierInNewRunspaces _rule;
+        
+        private readonly string _analyzedFilePath;
+        
+        public SyntaxCompatibilityVisitor(UseUsingScopeModifierInNewRunspaces rule, string analyzedScriptPath)
+        {
+            _diagnosticAccumulator = new List<DiagnosticRecord>();
+            _rule = rule;
+            _analyzedFilePath = analyzedScriptPath;
+        }
+
+        /// <summary>
+        /// GetDiagnosticRecords: Retrieves all Diagnostic Records that were generated during visiting
+        /// </summary>
+        public IEnumerable<DiagnosticRecord> GetDiagnosticRecords()
+        {
+            return _diagnosticAccumulator;
+        }
+
+        /// <summary>
+        /// VisitScriptBlockExpression: When a ScriptBlockExpression is visited, see if it belongs to a command that needs its variables
+        /// prefixed with the 'Using' scope modifier. If so, analyze the block and generate diagnostic records for variables where it is missing.
+        /// </summary>
+        /// <param name="scriptBlockExpressionAst"></param>
+        /// <returns>
+        /// AstVisitAction.Continue or AstVisitAction.SkipChildren, depending on what we found. Diagnostic records are saved in `_diagnosticAccumulator`.
+        /// </returns>
+        public override AstVisitAction VisitScriptBlockExpression(ScriptBlockExpressionAst scriptBlockExpressionAst)
+        {
+            if (!(scriptBlockExpressionAst.Parent is CommandAst commandAst))
+            {
+                return AstVisitAction.Continue;
+            }
+
+            var cmdName = commandAst.GetCommandName();
+            if (cmdName == null)
+            {
+                // Skip for situations where command name cannot be resolved like `& $commandName -ComputerName -ScriptBlock { $foo }` 
+                return AstVisitAction.Continue;
+            }
+            
+            var scriptBlockParameterAst = commandAst.CommandElements[
+                    commandAst.CommandElements.IndexOf(scriptBlockExpressionAst) - 1] as CommandParameterAst;
+
+            if (IsInlineScriptBlock(cmdName) || 
+                IsJobScriptBlock(cmdName, scriptBlockParameterAst) ||

```

Comment:
This looks for an explicit parameter name, but `Start-Job { <# do stuff #> }` seems quite common. Is it worth looking for explicit parameter?

Reply:
I think it is, yes. I see the comment explaining the why of this was lost in the refactor. For `Start-Job`, there can be a situation where there are _two_ ScriptPlocks passed in as parameters. The other one being `-InitializationScript`, which does not support the `using` scope modifier. That is why we need this information. I can add in a comment in the code for this. 
Did I interpret your comment the right way?

Reply:
Ah! Yep I see now. Thanks for the explanation